src/options/Makefile.am

```diff
@@ -262,179 +108,44 @@ nodist_liboptions_la_SOURCES = \
-		base_options_template.h.sed \
-		"$<" \
-	) > "$@"
-
-summary.sed : mkoptions $(OPTIONS_OPTIONS_FILES)
-	$(AM_V_at)chmod +x @srcdir@/mkoptions
-	$(AM_V_GEN)(@srcdir@/mkoptions summary-sed \
-		$(OPTIONS_OPTIONS_FILES) \
-	) > summary.sed
-
-
-
-#   mkoptions apply-sed-to-template sed-file template-file
-options_holder.h : options_holder_template.h options_holder_template.h.sed summary.sed mkoptions $(OPTIONS_HEADS)
-	$(AM_V_at)chmod +x @srcdir@/mkoptions
-	$(AM_V_GEN)(@srcdir@/mkoptions apply-sed-files-to-template \
-		$(if $(COVERAGE_ON), @abs_srcdir@, @srcdir@)/options_holder_template.h \
-		@builddir@/options_holder_template.h.sed \
-		summary.sed \
-	) > "$@"
-
-# Make sure not to match with "options.cpp" too.
-%_options.cpp: %_options.sed %_options.h mkoptions options_holder.h base_options_template.cpp base_options_template.cpp.sed
-	$(AM_V_at)chmod +x @srcdir@/mkoptions
-	$(AM_V_GEN)(@srcdir@/mkoptions apply-sed-files-to-template \
-		$(if $(COVERAGE_ON), @abs_srcdir@, @srcdir@)/base_options_template.cpp \
-		base_options_template.cpp.sed \
-		"$<" \
-	) > "$@"
-
-
-
-
-#   mkoptions apply-sed-to-template sed-file template-file
-options.cpp : options_template.cpp options_template.cpp.sed mkoptions summary.sed $(OPTIONS_HEADS) options_holder.h
-	$(AM_V_at)chmod +x @srcdir@/mkoptions
-	$(AM_V_GEN)(@srcdir@/mkoptions apply-sed-files-to-template \
-		$(if $(COVERAGE_ON), @abs_srcdir@, @srcdir@)/options_template.cpp \
-		@builddir@/options_template.cpp.sed \
-		summary.sed \
-	) > "$@"
-
-
-#   mkoptions apply-sed-to-template sed-file template-file
-options_get_option.cpp : options_get_option_template.cpp options_get_option_template.cpp.sed mkoptions summary.sed $(OPTIONS_HEADS)
-	$(AM_V_at)chmod +x @srcdir@/mkoptions
-	$(AM_V_GEN)(@srcdir@/mkoptions apply-sed-files-to-template \
-		$(if $(COVERAGE_ON), @abs_srcdir@, @srcdir@)/options_get_option_template.cpp \
-		@builddir@/options_get_option_template.cpp.sed \
-		summary.sed \
-	) > "$@"
-
-options_set_option.cpp : options_set_option_template.cpp options_set_option_template.cpp.sed mkoptions summary.sed $(OPTIONS_HEADS)
-	$(AM_V_at)chmod +x @srcdir@/mkoptions
-	$(AM_V_GEN)(@srcdir@/mkoptions apply-sed-files-to-template \
-		$(if $(COVERAGE_ON), @abs_srcdir@, @srcdir@)/options_set_option_template.cpp \
-		@builddir@/options_set_option_template.cpp.sed \
-		summary.sed \
-	) > "$@"
-
-
-
-$(DOCUMENTATION_FILES) : % : %_template %_template.sed mkoptions summary.sed
-#	echo "$<"
-#	echo "$@"
-	$(AM_V_at)chmod +x @srcdir@/mkoptions
-	$(AM_V_GEN)(@srcdir@/mkoptions apply-sed-files-to-template \
-		"$<" \
-		"$<".sed \
-		@builddir@/summary.sed \
-	) > "$@"
-
-
+$(CPP_TEMPLATE_FILES):;
 
-#options-stamp: options_holder_template.h options_template.cpp smt_options_template.cpp base_options_template.h base_options_template.cpp mkoptions $(OPTIONS_FILE_SRCS)
+# All source/doc files are generated in one pass with rule options.cpp
+$(OPTIONS_CPPS) $(OPTIONS_HEADS) $(DOCUMENTATION_FILES): options.cpp
```

Comment:
Won't this trigger a global re-compilation of all option headers on any option change? I am fine w/ this given the simplifications it yields. Have you asked Clark/Andy about this?

Reply:
mkoptions.py only updates files if they have changed, recompilation only happens for files that really changed.